Sources/Load.php

```diff
@@ -528,9 +529,9 @@ function loadUserSettings()
 			// As much as the password should be right, we can assume the integration set things up.
 			if (!empty($already_verified) && $already_verified === true)
 				$check = true;
-			// SHA-512 hash should be 128 characters long.
-			elseif (strlen($password) == 128)
-				$check = hash_salt($user_settings['passwd'], $user_settings['password_salt']) == $password;
+			// Password hash should be 40 characters long.
```

Comment:
With Sha3-512 and sha512, are those not 128 lengths?  Or am I missing something else that trims them back to 40.. Does the hmac do that?

Reply:
Good catch. Will fix shortly.